internal/provider/data_source_secure_connect_bundle_url.go

```diff
@@ -40,18 +42,36 @@ func dataSourceSecureConnectBundleURLRead(ctx context.Context, d *schema.Resourc
 
 	databaseID := d.Get("database_id").(string)
 
-	resp, err := client.GenerateSecureBundleURLWithResponse(ctx, astra.DatabaseIdParam(databaseID))
-	if err != nil {
-		return diag.FromErr(err)
-	}
+	var credsURL *astra.CredsURL
+	if err := resource.RetryContext(ctx, time.Minute, func() *resource.RetryError {
```

Comment:
Would it be worth breaking the retry out into it's own function for a) readability and b) unit testing with a mock astraClient?

Reply:
Good idea, done!